nixos/lib/default.nix

```diff
@@ -0,0 +1,47 @@
+let
+  # The warning is in a top-level let binding so it is only printed once.
+  experimentalWarning = warn "lib.nixos.evalModules is experimental and subject to change. See nixos/lib/default.nix" null;
+  inherit (nonExtendedLib) warn;
+  nonExtendedLib = import ../../lib;
```

Comment:
would turn that order around to make it a little more readable.

Reply:
This is evaluation order and each line follows step by step from the requirement outlined in the comment.
If I change the order around, the comment comes almost last, or is detached from its context.
Does that change your opinion?
Honest question because I have very little authority when it comes to readability of my own code of course.

Reply:
honestly we just prefer things to be defined lexically before their first use. your choice is just as valid as ours would be. :)

Reply:
I generally default to that as well. It's quite common in strict languages as well, to say the least ;)